packages/editor/src/components/post-transform-panel/hooks.js

```diff
@@ -84,31 +84,22 @@ function preparePatterns( patterns, template, currentThemeStylesheet ) {
 }
 
 export function useAvailablePatterns( template ) {
-	const { blockPatterns, restBlockPatterns, currentThemeStylesheet } =
-		useSelect( ( select ) => {
-			const { getSettings } = unlock( select( editSiteStore ) );
-			const settings = getSettings();
+	const { blockPatterns, currentThemeStylesheet } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
 
-			return {
-				blockPatterns:
-					settings.__experimentalAdditionalBlockPatterns ??
-					settings.__experimentalBlockPatterns,
-				restBlockPatterns: select( coreStore ).getBlockPatterns(),
-				currentThemeStylesheet:
-					select( coreStore ).getCurrentTheme().stylesheet,
-			};
-		}, [] );
+		return {
+			blockPatterns: getSettings().__experimentalBlockPatterns,
```

Comment:
I'm not entirely sure that the way I'm retrieving the patterns here is similar to what we had before. Also not sure how to test this panel properly yet.

Reply:
Also I'm guessing this PR brings this panel to the post editor (where it was not present)

Reply:
There should be a simple way to get the patterns. Each similar implementation in the core has to deal with extra logic, increasing the chances of missing something.

@ellatrix, you recently worked on refactoring pattern fetching logic. Is there a better way to get all the patterns?

Reply:
For now, I just restored the code that we had in the site editor to avoid impact but I agree that overall, we need to have a better story about retrieving patterns.